react/ShippingOptionList.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useMemo, useState } from 'react'
+import { DeliveryOption } from 'vtex.checkout-graphql'
+import { Alert } from 'vtex.styleguide'
+import { ListGroup } from 'vtex.checkout-components'
+import { getFastestSla, getCheapestSla } from '@vtex/estimate-calculator'
+import { FormattedMessage } from 'react-intl'
+
+import ShippingOption from './ShippingOption'
+
+interface Props {
+  deliveryOptions: DeliveryOption[]
+  onDeliveryOptionSelected?: (id: string) => void
+  onDeliveryOptionDeselected?: (id: string) => void
+}
+
+const ShippingOptionList: React.FC<Props> = ({
+  deliveryOptions,
+  onDeliveryOptionSelected = () => {},
+  onDeliveryOptionDeselected = () => {},
+}) => {
+  const handleDeliveryOptionSelect = (deliveryOption: DeliveryOption) => {
+    onDeliveryOptionSelected(deliveryOption.id!)
+  }
+
+  const handleDeliveryOptionDeselect = (deliveryOption: DeliveryOption) => {
+    onDeliveryOptionDeselected(deliveryOption.id!)
+  }
+
+  const fastestOption = useMemo(() => {
+    return getFastestSla(
+      deliveryOptions.map(({ id, estimate, price }) => ({
+        id: id!,
+        shippingEstimate: estimate!,
+        price,
+      }))
+    )
+  }, [deliveryOptions])
+
+  const cheapestOption = useMemo(() => {
+    return getCheapestSla(
+      deliveryOptions.map(({ id, estimate, price }) => ({
+        id: id!,
+        shippingEstimate: estimate!,
+        price,
+      }))
+    )
+  }, [deliveryOptions])
+
+  const selectedDeliveryOptions = useMemo(() => {
+    return deliveryOptions.filter(({ isSelected }) => isSelected)
+  }, [deliveryOptions])
+
+  const [showSelectedDeliveryOption, setShowSelectedDeliveryOption] = useState(
+    true
+  )
+
+  return deliveryOptions.length > 0 ? (
+    showSelectedDeliveryOption ? (
+      <>
+        <p className="mt0 mb5 fw4 f4 lh-copy">
+          <FormattedMessage id="store/checkout.shipping.selectedDeliveryOptionsLabel" />
+        </p>
+
+        <ListGroup>
+          {selectedDeliveryOptions.map((deliveryOption) => (
+            <ShippingOption
+              key={deliveryOption.id}
```

Comment:
question: not sure if that matters here, but why not using the non-null assertion here as well? Same for line 83.

Reply:
because the types for a component key allows null values as well

Reply:
sure, but I just wondered if it would be good to raise a warning in this case, as it isn't an expected behavior

Reply:
I don' think the delivery option id can _actually_ be `null`, maybe the typings on our GraphQL are incorrect